enumerables.rb

```diff
@@ -0,0 +1,153 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield to_a[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield to_a[i], i
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    result = []
+    to_a.my_each { |item| result << item if yield(item) }
+    result
+  end
+
+  def my_all?
+    to_a.my_each do |i|
+      return false if yield(i) == false
+      else
+        return true
+      end
+    end
+  end
+
+  def my_any?
+    return to_enum(:my_any?) unless block_given?
+
+    i = 0
+    ans = false
+    to_a.length.times do
+      ans = yield to_a[i]
+
+      break if ans == true
+
+      i += 1
+    end
+    ans
+  end
+
+  def my_none?
+    return to_enum(:my_none?) unless block_given?
+
+    i = 0
+    ans = false
+    to_a.length.times do
+      ans = yield to_a[i]
+
+      break if ans == true
+
+      i += 1
+    end
+    !ans
+  end
+
+  def my_count(number = nil)
+    count = 0
+    if number.nil?
+      count = to_a.length
+    else
+
+      my_each { |i| count += 1 if number == i }
+
+      print count
+    end
+  end
+
+  def my_map(proc = nil)
+    result = []
+
+    to_a.my_each { |item| result << item if proc.call(item) }
+    result
+  end
```

Comment:
***my_map***

I will be using these variables for the next suggestions:

```
array = [2, 5, 9, 7, 2, 2]
my_proc = proc { |num| num > 10 }
``` 

- [ ] Returns a new array with the results of running a given block once for every element in enum.

```
      array.my_map { |num| num < 10 }
      
      NoMethodError:
        undefined method `call' for nil:NilClass
      # ./enumerables.rb:84:in `block in my_map'
      # ./enumerables.rb:7:in `my_each'
      # ./enumerables.rb:84:in `my_map'
```

- [ ] Executes only the proc when both a block and a proc are given and returns a new array with the results of running the proc once for every element in enum.

```
      array.my_map(my_proc) { |num| num < 10 }
      
        Expected: [false, false, false, false, false, false]
        Got: []
```